src/vm/mlinfo.cpp

```diff
@@ -5019,7 +5019,10 @@ void ArrayMarshalInfo::InitElementInfo(CorNativeType arrayNativeType, MarshalInf
 {
     CONTRACT_VOID
     {
-        STANDARD_VM_CHECK;
+        THROWS;
+        GC_TRIGGERS;
+        MODE_ANY;
```

Comment:
Why did you need to change these to `MODE_ANY`?

I think you should keep these as `STANDARD_VM_CHECK` and switch to preemptive mode around the entrypoint to the field loader.

It is a GC deadlock bug farm to have large amounts of code run under `MODE_ANY`.

Reply:
I believe I changed them to avoid needing to do GC-mode switching in `marshalnative.cpp` since it seemed that everything they called was also `MODE_ANY`. I'll change `marshalnative.cpp` to switch to preemptive GC when calling into the interop subsystem.